harness/determined/cli/checkpoint.py

```diff
@@ -7,6 +7,7 @@
 from determined.common.api import authentication, bindings
 from determined.common.declarative_argparse import Arg, Cmd
 from determined.common.experimental import Determined
+from determined.common.experimental.checkpoint import DownloadMode
```

Comment:
Defining `DownloadMode` where you have it is fine, but can you also import it into `determined/experimental/client.py` and import it from there?

I know that this cli code is messy, but that's the path it _should_ be imported from, both by internal code and end users.  This `det.common.experimental.checkpoint` path is deprecated, it just hasn't been removed yet.